src/main/java/com/minecolonies/util/BlockUtils.java

```diff
@@ -106,4 +107,140 @@ public static boolean isWater(IBlockState iBlockState)
         return Objects.equals(iBlockState, Blocks.water.getDefaultState())
                || Objects.equals(iBlockState, Blocks.flowing_water.getDefaultState());
     }
+
+    /**
+     * Gets an item from a block.
+     * @param block input block.
+     * @return output item.
+     */
+    public static Item getItemFromBlock(Block block)
+    {
+        Item item = null;
+
+        if(block instanceof BlockBanner)
+        {
+            item = Items.banner;
+        }
+        else if(block instanceof BlockBed)
+        {
+            item = Items.bed;
+        }
+        else if(block instanceof BlockBrewingStand)
+        {
+            item = Items.brewing_stand;
+        }
+        else if(block instanceof BlockCake)
+        {
+            item = Items.cake;
+        }
+        else if(block instanceof BlockCauldron)
+        {
+            item = Items.cauldron;
+        }
+        else if(block instanceof BlockCocoa)
+        {
+            item = Items.dye;
+        }
+        else if(block instanceof BlockCrops)
+        {
+            if(block instanceof BlockCarrot)
+            {
+                item = Items.carrot;
+            }
+            else if(block instanceof BlockPotato)
+            {
+                item = Items.potato;
+            }
+            else
+            {
+                item = Items.wheat_seeds;
+            }
+        }
+        else if(block instanceof BlockDaylightDetector)
+        {
+            item = Item.getItemFromBlock(Blocks.daylight_detector);
+        }
+        else if(block instanceof BlockDoor)
+        {
+            item = block == Blocks.iron_door ? Items.iron_door
```

Comment:
LGTM

Grüße, 
Marvin 

Am 17.08.2016 07:56 schrieb Minecolonies Sonar Checker <notifications@github.com>:

In src/main/java/com/minecolonies/util/BlockUtils.java:

> + else if(block instanceof BlockPotato) > + { > + item = Items.potato; > + } > + else > + { > + item = Items.wheat_seeds; > + } > + } > + else if(block instanceof BlockDaylightDetector) > + { > + item = Item.getItemFromBlock(Blocks.daylight_detector); > + } > + else if(block instanceof BlockDoor) > + { > + item = block == Blocks.iron_door ? Items.iron_door 

  Reduce the number of conditional operators (6) used in the expression (maximum allowed 4).  

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or mute the thread.